tmexp/create_bow.py

```diff
@@ -0,0 +1,218 @@
+from collections import Counter
+import copy
+import logging
+import os
+import pickle
+from typing import Any, Dict, List, Optional
+
+import tqdm
+
+from .utils import (
+    check_exists,
+    check_remove_file,
+    create_directory,
+    create_language_list,
+)
+
+DIFF_MODEL = "diff"
+HALL_MODEL = "hall"
+SEP = ":"
+
+
+def check_fraction(fraction: float, arg_name: str) -> None:
+    if not (0 <= fraction <= 1):
+        raise RuntimeError(
+            "Argument '%s' must be in the range [0, 1], aborting." % arg_name
+        )
+
+
+def create_bow(
+    input_path: str,
+    output_dir: str,
+    dataset_name: str,
+    langs: Optional[List[str]],
+    exclude_langs: Optional[List[str]],
+    features: List[str],
+    force: bool,
+    topic_model: str,
+    min_word_frac: float,
+    max_word_frac: float,
+    log_level: str,
+) -> None:
+    logger = logging.getLogger("create_bow")
+    logger.addHandler(logging.StreamHandler())
+    logger.setLevel(log_level)
+
+    check_exists(input_path)
+    create_directory(output_dir, logger)
+    if dataset_name == "":
+        dataset_name = topic_model
+    words_output_path = os.path.join(output_dir, "vocab." + dataset_name + ".txt")
+    check_remove_file(words_output_path, logger, force)
+    docword_output_path = os.path.join(output_dir, "docword." + dataset_name + ".txt")
+    check_remove_file(docword_output_path, logger, force)
+    docs_output_path = os.path.join(output_dir, "docs." + dataset_name + ".txt")
+    check_remove_file(docs_output_path, logger, force)
+    check_fraction(min_word_frac, "min-word-frac")
+    check_fraction(max_word_frac, "max-word-frac")
+
+    logger.info("Reading pickled dict from '%s' ..." % input_path)
+    with open(input_path, "rb") as _in:
+        input_dict = pickle.load(_in)
+
+    logger.info("Computing bag of words ...")
+    langs = create_language_list(langs, exclude_langs)
+    all_refs = sorted([ref for ref in input_dict["files_info"]])
```

Comment:
`sorted` accepts a generator, no need to construct the intermediate list. Also, we don't do anything except iterating on the dict keys so we can just give it to sorted directly:

```suggestion
    all_refs = sorted(input_dict["files_info"])
```

Reply:
done